omnigan/trainer.py

```diff
@@ -103,6 +101,23 @@ def __init__(self, opts, comet_exp=None, verbose=0, device=None):
             self.exp = comet_exp
         self.domain_labels = {"s": 0, "r": 1}
 
+        if self.opts.train.amp:
+            optimizers = [
+                self.opts.gen.opt.optimizer.lower(),
+                self.opts.dis.opt.optimizer.lower(),
+                self.opts.classifier.opt.optimizer.lower(),
+            ]
+            if "extraadam" in optimizers:
+                raise ValueError(
+                    "AMP does not work with ExtraAdam ({})".format(optimizers)
+                )
+            self.grad_scaler_d = GradScaler()
+            self.grad_scaler_g = GradScaler()
+            self.grad_scaler_c = GradScaler()
+
+    def mask(self, z):
```

Comment:
I don't see where this function is used, maybe you forgot about it?:) Because it's a good idea, we could change all lines in trainer.py with sigmoid(self.G.decoders["m"](z)) to self.mask(z), and also use it in infer.py (do trainer.mask(z) instead of sigmoid(model.decoders["m"](z))).

Reply:
Yes I was thinking about inference but you're right we could use it elsewhere